usr/src/cmd/tar/tar.c

```diff
@@ -8271,7 +8271,7 @@ read_xattr_hdr(attr_data_t **attrinfo)
 	(void) sscanf(xattrp->h_namesz, "%7d", &namelen);
 	if (link_len > 0)
 		xattr_linkp = (struct xattr_buf *)
-		    ((int)xattrp + (int)comp_len);
+		    ((uintptr_t)xattrp + (int)comp_len);
```

Comment:
I think this should just be

```suggestion
    xattr_linkp = (struct xattr_buf *)(xattrp + comp_len);
```